editor/arithmetic.py

```diff
@@ -158,12 +158,24 @@ def execute_simple(self, operand: Expression) -> Expression:
         return bools[operand is SingletonFalse]
 
 
+@global_attr("eqv?")
+class IsEqv(BuiltIn):
+    def execute_evaluated(self, operands: List[Expression], frame: Frame):
+        verify_exact_callable_length(self, 2, len(operands))
+        if all(isinstance(x, ValueHolder) for x in operands):
+            return bools[operands[0].value == operands[1].value]
+        return bools[operands[0] is operands[1]]
+
+
 @global_attr("eq?")
 class IsEq(BuiltIn):
     def execute_evaluated(self, operands: List[Expression], frame: Frame):
         verify_exact_callable_length(self, 2, len(operands))
         if all(isinstance(x, ValueHolder) for x in operands):
-            return bools[operands[0].value == operands[1].value]
+            if isinstance(operands[0], Number):
+                return bools[operands[0] is operands[1]]
+            else:
+                return bools[operands[0].value == operands[1].value]
```

Comment:
I'm still confused about eq, it's not is for non-numbers too?

Reply:
It is, on line 179. But it's not `is` for Symbols or booleans (or other `ValueHolders`, like strings)

Reply:
I don't know why it is like this, but Prof. Hilfinger wanted it this way.